include/remill/Arch/Runtime/Intrinsics.h

```diff
@@ -303,4 +305,111 @@ __remill_write_io_port_16(Memory *, addr_t, uint16_t);
 [[gnu::used, gnu::const]] extern Memory *
 __remill_write_io_port_32(Memory *, addr_t, uint32_t);
 
+// More specific hyper calls.
+[[gnu::used, gnu::const]] extern Memory *__remill_x86_set_segment_es(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *__remill_x86_set_segment_ss(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *__remill_x86_set_segment_ds(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *__remill_x86_set_segment_fs(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *__remill_x86_set_segment_gs(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *__remill_x86_set_debug_reg(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_x86_set_control_reg_0(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_x86_set_control_reg_1(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_x86_set_control_reg_2(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_x86_set_control_reg_3(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_x86_set_control_reg_4(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *__remill_amd64_set_debug_reg(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_amd64_set_control_reg_0(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_amd64_set_control_reg_1(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_amd64_set_control_reg_2(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_amd64_set_control_reg_3(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_amd64_set_control_reg_4(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_amd64_set_control_reg_8(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_aarch64_emulate_instruction(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_aarch32_emulate_instruction(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_aarch32_check_not_el2(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_sparc_set_asi_register(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_sparc_unimplemented_instruction(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_sparc_unhandled_dcti(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *
+__remill_sparc_window_underflow(Memory *);
+
+[[gnu::used, gnu::const]] extern Memory *__remill_sparc_trap_cond_a(Memory *);
```

Comment:
I think a `bool` or something like it should probably be passed in here as another parameter. It feels like the trapping conditions themselves could be elided, or ordered before/after this intrinsic call.

Reply:
Perhaps I can pass in the value of `HYPER_CALL_VECTOR`?

https://github.com/lifting-bits/remill/blob/master/lib/Arch/SPARC32/Semantics/TRAP.cpp#L32-L38

It's a `uint32_t` listed [here](https://github.com/lifting-bits/remill/blob/master/include/remill/Arch/Runtime/State.h#L34).

Reply:
Oh indeed! I was thinking this must be some kind of thing like "division by zero" with data-dependant conditions.